newIDE/app/src/VariablesList/index.js

```diff
@@ -254,43 +252,78 @@ export default class VariablesList extends React.Component<Props, State> {
     );
   }
 
-  render() {
+  onAddVariable = (variableName = false) => {
     const { variablesContainer } = this.props;
+    const variable = new gd.Variable();
+    variable.setString('');
+
+    const name = variableName
+      ? variableName
+      : newNameGenerator('Variable', name => variablesContainer.has(name));
+    variablesContainer.insert(name, variable, variablesContainer.count());
+    return variable;
+  };
+
+  render() {
+    const { variablesContainer, objectVariables } = this.props;
     if (!variablesContainer) return null;
 
+    /// map all instance variables with the same name as object variables
+    const objVariablesTree = objectVariables
```

Comment:
You consider objectVariables as mandatory (which will surely make Flow scream ;)), meaning that any VariablesList not passing objectVariables will crash - you should consider it's optional